src/Build/BackEnd/Node/ServerNamedMutex.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+
+namespace Microsoft.Build.Execution
+{
+    internal sealed class ServerNamedMutex : IDisposable
+    {
+        private readonly Mutex _serverMutex;
+
+        public bool IsDisposed { get; private set; }
+
+        public bool IsLocked { get; private set; }
+
+        public ServerNamedMutex(string mutexName, out bool createdNew)
+        {
+            _serverMutex = new Mutex(
+                initiallyOwned: true,
+                name: mutexName,
+                createdNew: out createdNew);
+
+            if (createdNew)
+            {
+                IsLocked = true;
+            }
+        }
+
+        internal static ServerNamedMutex OpenOrCreateMutex(string name, out bool createdNew)
+        {
+            // TODO: verify it is not needed anymore
+            // if (PlatformInformation.IsRunningOnMono)
+            // {
+            //     return new ServerFileMutexPair(name, initiallyOwned: true, out createdNew);
+            // }
+            // else
+
+            return new ServerNamedMutex(name, out createdNew);
+        }
+
+        public static bool WasOpen(string mutexName)
```

Comment:
When you acquire kernel object like Mutex it has to be disposed to return its resources back to OS. If we had not dispose it it would be disposed/collected by finalizer anyway.

NOTE: This logic and code has been cloned from https://github.com/dotnet/roslyn/blob/579732c393b69462f489be05019ce2eaa329e648/src/Compilers/Shared/BuildServerConnection.cs#L844
as it was used for VBCSCompiler which solves similar problem like MSBuild Server. I believe this code is correct and it was battle tested for years.

Reply:
@rokonec is correct, that is why this value is disposed